spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -297,7 +300,12 @@ protected ConnectionFactory createInstance() throws Exception {
 	 */
 	protected void setUpSSL() throws Exception {
 		if (this.sslPropertiesLocation == null) {
-			this.connectionFactory.useSslProtocol();
+			if (this.sslAlgorithmSet) {
+				this.connectionFactory.useSslProtocol(this.sslAlgorithm);
+			}
+			else {
+				this.connectionFactory.useSslProtocol();
```

Comment:
How about to always propagate our `this.sslAlgorithm`, even if default, to override that `TLSv1` in the `connectionFactory`? At least for our `1.5`. And notify RabbitMQ team to change their default value.

CC @michaelklishin